test/python/test_prefetched.py

```diff
@@ -0,0 +1,160 @@
+# ==============================================================================
+#  Copyright 2018-2019 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+# ==============================================================================
+"""nGraph TensorFlow bridge prefetch test
+
+"""
+import sys
+import pytest
+import getpass
+import tensorflow as tf
+from tensorflow.python.framework import dtypes
+tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
+
+import ngraph_bridge
+
+import numpy as np
+from common import NgraphTest
+
+import warnings
+warnings.filterwarnings('ignore', category=FutureWarning)
+
+import ngraph_bridge
+
+
+class TestPrefetched(NgraphTest):
+
+    def build_data_pipeline(self, input_array, map_function, batch_size):
+        dataset = (tf.data.Dataset.from_tensor_slices(
+            (tf.constant(input_array)
+            )).map(map_function).batch(batch_size).prefetch(1))
+
+        iterator = dataset.make_initializable_iterator()
+        data_to_be_prefetched_and_used = iterator.get_next()
+        return data_to_be_prefetched_and_used, iterator
+
+    def build_model1(self, input_array, c1, c2):
+        # Convert the numpy array to TF Tensor
+        input_f = tf.cast(input_array, tf.float32)
+
+        # Define the Ops
+        pl1 = tf.placeholder(dtype=dtypes.int32)
+        pl1_f = tf.cast(pl1, tf.float32)
+        pl2 = tf.placeholder(dtype=dtypes.int32)
+        pl2_f = tf.cast(pl2, tf.float32)
+
+        mul = tf.compat.v1.math.multiply(input_f, c1)
+        add = tf.compat.v1.math.add(mul, pl2_f)
+        add2 = add + pl1_f
+        output = add2 - c2
+        return output, pl1, pl2
+
+    def build_model2(self, input_array, c1, c2):
```

Comment:
I was trying to play with the input indexes. ie in model1 input index0 is prefetched and model2 index 1 is prefetched. i just dumped the graphs and TF always creates the graph with index 0 as prefetched.  I will get rid of model 2/test2